tools/secret-rotation/Azure.Sdk.Tools.SecretRotation.Cli/Commands/StatusCommand.cs

```diff
@@ -17,13 +20,130 @@ public StatusCommand() : base("status", "Show secret rotation status")
+                builder.AppendLine($"    RequiresRevocation: {status.RequiresRevocation}");
+                builder.AppendLine($"    Exception: {status.Exception?.Message}");
+
+                logger.LogDebug(builder.ToString());
+            }
+
+
+            statuses.Add((plan, status));
         }
+
+        var errored = statuses.Where(x => x.Status.Exception is not null).ToArray();
+        var expired = statuses.Except(errored).Where(x => x.Status is { Expired: true }).ToArray();
+        var expiring = statuses.Except(errored).Where(x => x.Status is { Expired: false, ThresholdExpired: true }).ToArray();
+        var upToDate = statuses.Except(errored).Where(x => x.Status is { Expired: false, ThresholdExpired: false }).ToArray();
+
+        var statusBuilder = new StringBuilder();
+
+        void AppendStatusSection(IList<(RotationPlan Plan, RotationPlanStatus Status)> sectionStatuses, string header)
+        {
+            if (!sectionStatuses.Any())
+            {
+                return;
+            }
+
+            statusBuilder.AppendLine();
+            statusBuilder.AppendLine(header);
+            foreach ((RotationPlan plan, RotationPlanStatus status) in sectionStatuses)
+            {
+                foreach (string line in GetPlanStatusLine(plan, status).Split("\n"))
+                {
+                    statusBuilder.Append("  ");
+                    statusBuilder.AppendLine(line);
+                }
+            }
+        }
+
+        AppendStatusSection(expired, "Expired:");
+        AppendStatusSection(expiring, "Expiring:");
+        AppendStatusSection(upToDate, "Up-to-date:");
+        AppendStatusSection(errored, "Error reading plan status:");
+
+        logger.LogInformation(statusBuilder.ToString());
+
+        if (expired.Any())
+        {
+            invocationContext.ExitCode = 1;
+        }
+    }
+
+    private static string GetPlanStatusLine(RotationPlan plan, RotationPlanStatus status)
+    {
+        if (status.Exception != null)
+        {
+            return $"{plan.Name}:\n  {status.Exception.Message}";
+        }
+
+        DateTimeOffset? expirationDate = status.ExpirationDate;
+
+        DateTimeOffset now = DateTimeOffset.UtcNow;
+
+        string expiration = expirationDate.HasValue
+            ? $"{FormatTimeSpan(expirationDate.Value.Subtract(now))}"
+            : "No expiration date";
+
+        return $"{plan.Name} - {expiration} / ({FormatTimeSpan(plan.RotationPeriod)} @ {FormatTimeSpan(plan.RotationThreshold)})";
+    }
+
+    private static string FormatTimeSpan(TimeSpan timeSpan)
+    {
+        if (timeSpan == TimeSpan.Zero)
+        {
+            return "0d";
+        }
+
+        StringBuilder builder = new StringBuilder();
+
+        if (timeSpan.Days > 0)
+        {
+            builder.Append(timeSpan.Days);
+            builder.Append('d');
+        }
+
+        if (timeSpan.Days < 2 && timeSpan.TotalDays - timeSpan.Days > 0)
```

Comment:
Do we not print anything if time > 2?

Reply:
We don't add a time component to timeSpan > 2d.

48:01:00 => '2d'
48:00:00 => '2d'
47:59:00 =>  '1d 23h 59m'